addons/ailogistics/functions/fnc_logisticsMain.sqf

```diff
@@ -0,0 +1,110 @@
+#include "script_component.hpp"
+/*
+    Author:
+        Bridg
+
+    Description:
+        Main function
+
+    Parameters:
+
+
+    Return value:
+        Nothing
+*/
+
+
+if (!isServer) exitWith {};
+
+(_this select 1) params ["_module"];
+
+
+// get all modules parameters
+private _destinations = synchronizedObjects _module;
+private _time = _module getVariable [QGVAR(logisticsTimeout),0];
+private _groupSize = _module getVariable [QGVAR(picketGroupSize),0];
+private _picketUnits = _module getVariable [QGVAR(picketUnitPoolString),[]];
+private _sideFromModule = _module getVariable [QGVAR(side),0];
+
+// get side then update the variable with an actual side, get the updated version
+if (_sideFromModule == 0) then {
+    _module setVariable [QGVAR(side),EAST];
+} else {
+    _module setVariable [QGVAR(side),INDEPENDENT];
+};
+private _side = _module getVariable [QGVAR(side),EAST];
+
+
+
+// establish the route
+private _agent = createAgent ["C_Man_1", getPos _module, [], 0, "NONE"];
+private _car = "B_Quadbike_01_F" createVehicle (getPos _module);
+_agent moveInDriver _car;
+
+_agent setVariable [QGVAR(module),_module];
+
+// workaround ai spawning at the end the route due to double execution bug of pathCalculated EH.
+// if this is called twice by sync-ing more than one destination module, it only picks route. Seems that you have to wait until the path calc's once before you can do it again.
+// if there is a way to do the above, one start module could support multiple destinations 1, would also require multi vehicle support
+
+_agent setDestination [(_destinations select 0), "LEADER PLANNED", true];
+
+_agent addEventHandler ["PathCalculated", {
+    params ["_agent", "_path"];
+    private _module = _agent getVariable [QGVAR(module),objNull];
+    if ((_module getVariable [QGVAR(routePickets),false]) == false) exitWith {};
```

Comment:
If this isn't being used when pickets are not setup, don't create the agent and calculate a path etc.

Move this agent creation and picket calulation to a new function, and only call it from here if `routePickets` is `true`.

Reply:
Removed path calc as a method entirely